### PR TITLE
feat: enhance documentation for desktop-only plugins

### PR DESCRIPTION
**Pull request by lucasfernog**
usage should actually be behind the #[cfg(desktop)] flag

### CODE REVIEW

File: src/content/docs/plugin/single-instance.mdx [hunk 1]
```diff
@@ -96,10 +96,15 @@ By default, when you initiate a new instance while the application is already ru
 use tauri::{AppHandle, Manager};
 
 pub fn run() {
-    tauri::Builder::default()
-        .plugin(tauri_plugin_single_instance::init(|app, args, cwd| {
+    let mut builder = ;tauri::Builder::default()
```
Comment (FabianLars) [inline src/content/docs/plugin/single-instance.mdx]:
```suggestion
    let mut builder = tauri::Builder::default();
```

Comment (lucasfernog) [reply src/content/docs/plugin/single-instance.mdx]: just checking if you're actually reviewing or just clicking 🤣

**Comment by FabianLars** [comment]
ngl looking at this i kindaaa want to change the platform specific plugins to be compilable on all platforms, then you wouldn't have to have that many cfg flags (unless you want maximum optimization ig). But that can be done later too if we actually want to at some point 🤷

**Review by FabianLars — APPROVED** [review]

**Comment by dreyfus92** [comment]
Thank you both! 😁

**Comment by lucasfernog** [comment]
> ngl looking at this i kindaaa want to change the platform specific plugins to be compilable on all platforms, then you wouldn't have to have that many cfg flags (unless you want maximum optimization ig). But that can be done later too if we actually want to at some point 🤷 

i remember chip also wanted to do something similar for the entire codebase.. have less cfg and more no-ops would be beneficial

**Pull request merged**